src/Stratis.Bitcoin.Features.Api/Program.cs

```diff
@@ -9,18 +11,26 @@ namespace Stratis.Bitcoin.Features.Api
 {
     public class Program
     {
-        public static void Main(string[] args)
-        {
-        }
-
-        public static IWebHost Initialize(IEnumerable<ServiceDescriptor> services, FullNode fullNode, ApiSettings apiSettings)
+        public static IWebHost Initialize(IEnumerable<ServiceDescriptor> services, FullNode fullNode,
+            ApiSettings apiSettings, ICertificateStore store, IWebHostBuilder webHostBuilder = null)
         {
             Guard.NotNull(fullNode, nameof(fullNode));
 
             Uri apiUri = apiSettings.ApiUri;
 
-            IWebHost host = new WebHostBuilder()
-                .UseKestrel()
+            X509Certificate2 certificate = apiSettings.UseHttps ? GetHttpsCertificate(apiSettings, store) : null;
+
+            webHostBuilder = webHostBuilder ?? new WebHostBuilder();
+
+            webHostBuilder
+                .UseKestrel(options =>
+                    {
+                        if (!apiSettings.UseHttps) return;
+                        options.Listen(
```

Comment:
please add a new line after `if (!apiSettings.UseHttps) return;`, I find it a bit too crammed like that, don't you?

Reply:
for me it is better in one line, this is really "read and forget", but I think most people are like you. Changed.